src/Buffer.java

```diff
@@ -27,13 +31,17 @@ synchronized String consume() {
             }
         }
         product = this.buffer.remove();
+        int result = -1;
+        //TODO:- Agregar la lógica para calcular el resultado de la cadena de scheme.
+        gui.completarTareaPorHacer();
+        gui.anadirTareaRealizada(id, product, result);
         Buffer.print("Consumer consumed: " + product);
         notify();
         
         return product;
     }
     
-    synchronized void produce(String product) {
+    synchronized void produce(String product, int ID) {

```

Comment:
```suggestion
    synchronized void produce(String product, int id) {
```
Cambiar a id para mantener la consistencia con la naming convention.